### ISSUE TITLE
YOLOv9/v10/v11 Detecting Butterfly as Person (False Positive) — How to Reduce False Detections?

### ISSUE THREAD

**Issue by harimukundh** [ultralytics/ultralytics#21423]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, I’m facing an issue with false detections while using pre-trained YOLO models. I tested the same image (attached below) of a butterfly with the following models:

YOLOv9

YOLOv10

YOLOv11

All of them are detecting the butterfly as a person with varying confidence scores (between 0.26 and 0.90). Here’s the sample code I used:

from ultralytics import YOLO
Path to image

image_path = “butterfly.jpg”
Load model

model_v10 = YOLO(‘yolov10s.pt’)
Run inference

results_v10 = model_v10(image_path)
Check and display results

if not results_v10[0].boxes or len(results_v10[0].boxes) == 0:
print(“YOLOv10: No objects detected.”)
else:
print(“YOLOv10: Objects detected.”)
results_v10[0].show()
Problem:

    The model detects the butterfly as a person.
    I want to avoid false detections like this, especially when the object is not present in the scene (like a person in this case).

My Questions:

    How can I reduce false detections in YOLO models for such cases?
    I don’t want to raise the confidence threshold above 0.4**, because I want to preserve valid lower-confidence detections. But I still want to reduce incorrect detections like a butterfly being marked as a person. What’s the best way to achieve this?`

![Image](https://github.com/user-attachments/assets/7ac1deb7-67fb-4fde-88cc-3c5537dc5038)

`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @harimukundh, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) to help us debug it. Please include the exact model weights, your inference code, and the image causing the false positive (thank you for sharing details and your code snippet—if possible, please also attach your image directly for full reproducibility).

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response. An Ultralytics engineer will review your issue and provide further assistance soon! 🦋

**Comment by Y-T-G** [comment]
You can try running prediction with  `augment=True`

**Comment by atalaydenknalbant** [comment]
![Image](https://github.com/user-attachments/assets/17d1266b-38ef-4415-8622-3d9b9fe4d25a)

This is an out of distribution detection problem. The core issue is not a failure of the model but a consequence of its training. The pretrained YOLO models you are using were trained on the COCO dataset which contains only 80 specific object classes and a butterfly is not one of them. You can find the exact list of these 80 classes in the dataset configuration file at [ultralytics/cfg/datasets/coco.yaml](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/coco.yaml) within the library's repository.

When the model is forced to classify an object outside its knowledge base it attempts to match the object's features to the closest class it knows. A butterfly's shape symmetry and patterns can trigger feature detectors associated with known objects. Smaller models have less nuanced feature extractors and might vaguely match the butterfly's upright symmetrical shape to a person's silhouette resulting in a low confidence detection as you observed.

<img width="227" height="222" alt="Image" src="https://github.com/user-attachments/assets/9e2bf2bd-43e0-40ec-b581-caeae49bd2ee" />

Larger more powerful models have learned more complex and abstract features. For example the image I provided showing a detection of "umbrella" with 0.88 confidence is from a larger model yolo11x which matched the wing's canopy like shape and radial pattern to its internal representation of an umbrella leading to a high confidence yet incorrect classification.

To solve this while respecting your constraint of keeping the confidence threshold below 0.4 you can either fine tune the model or train from scratch using yolo.yaml files.(yolo11n.yaml for example)

The most accurate long term solution is to fine tune a pre trained model on a custom dataset by first gathering and annotating images that include positive examples of butterflies in various conditions and negative examples like birds umbrellas or background-only images to teach the model what not to detect then labeling all relevant objects with bounding boxes including butterflies and selected COCO classes and finally using the Ultralytics framework to train the YOLO model so it learns to distinguish butterflies from other classes and reduces false positives across all categories or alternatively you can search for small [VLM models](https://www.ultralytics.com/blog/understanding-vision-language-models-and-their-applications) that covers zero shot object detection which might better suit your requirements if you prefer to avoid collecting or annotating data.

**Comment by glenn-jocher** [comment]
@atalaydenknalbant provided an excellent analysis of the out-of-distribution detection issue. As a quick additional approach, you could implement post-processing filters based on detection geometry - for instance, filtering out "person" detections with unusual aspect ratios or sizes that don't match typical human proportions. You could also consider using YOLO-World or other open-vocabulary models that can handle objects beyond the COCO classes without requiring custom training.
